### PR TITLE
New URL format for analysis message docs

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/25852

### CODE REVIEW

File: galley/pkg/config/analysis/diag/message_test.go [hunk 1]
```diff
@@ -67,6 +67,6 @@ func TestMessage_JSON(t *testing.T) {
 	m := NewMessage(mt, &resource.Instance{Origin: testOrigin{name: "toppings/cheese", ref: testReference{"path/to/file"}}}, "Feta")
 
 	j, _ := json.Marshal(&m)
-	g.Expect(string(j)).To(Equal(`{"code":"IST-0042","documentation_url":"https://istio.io/docs/reference/config/analysis/IST-0042"` +
+	g.Expect(string(j)).To(Equal(`{"code":"IST-0042","documentation_url":"https://istio.io/docs/reference/config/analysis/ist-0042/"` +
```
Comment (frankbu) [inline galley/pkg/config/analysis/diag/message_test.go:70]: What is this? There doesn't seem to be a code IST-0042 in the docs. Also all the other ones have no dash, i.e., they are all of the form `analysis/istNNNN`

Comment (frankbu) [reply galley/pkg/config/analysis/diag/message_test.go:70]: Oh, I see it's just a unit test ... might want to take out the `-` anyway.

Comment (esnible) [reply galley/pkg/config/analysis/diag/message_test.go]: Fixed

**Pull request merged**